lcls_tools/common/image_processing/roi.py

```diff
@@ -0,0 +1,109 @@
+from abc import ABC, abstractmethod
+from typing import (
+    List,
+)
+import numpy as np
+from matplotlib import patches
+from pydantic import BaseModel, PositiveFloat, Field
+
+
+class ROI(BaseModel, ABC):
+    roi_type: str
+    center: List[PositiveFloat]
+
+    @abstractmethod
+    def crop_image(self, img, **kwargs) -> np.ndarray:
+        """crop image using ROI"""
+        pass
```

Comment:
I think using `...` here in favour of `pass` is preferred as it's an abstractmethod (even `pass` is an implementation)

https://docs.python.org/3/library/abc.html#abc.abstractmethod